uitools/cpp/Esri/ArcGISRuntime/Toolkit/OverviewMapController.cpp

```diff
@@ -0,0 +1,395 @@
+    underlying inset of the OverviewMap tool.
+
+    For the C++/Quick toolkit this is a MapQuickView, for the C++/Widgets toolkit 
+    this is a MapGraphicsView.
+   */
+  MapViewToolkit* OverviewMapController::insetView() const
+  {
+    return m_insetView;
+  }
+
+  /*!
+    \brief Returns the symbol used to represent the GeoView's current viewpoint.
+
+    If \l OverviewMapController::geoView is not set, then the symbol will 
+    return \c{nullptr} by default. 
+    
+    If \l OverviewMapController::geoView is set, and a custom symbol was not applied, 
+    then a default will be provided.
+
+    For MapView, the default is an empty SimpleFillSymbol with a 1 point red outline;
+    for SceneView, it will be a red, cross-hair SimpleMarkerSymbol.
+   */
+  Symbol* OverviewMapController::symbol() const
+  {
+    return m_reticle->symbol();
+  }
+
+  /*!
+    \brief Sets the symbol used to represent the GeoView's current viewpoint.
+   */
+  void OverviewMapController::setSymbol(Symbol* symbol)
+  {
+    if (m_reticle->symbol() == symbol)
+      return;
+
+    if (m_reticle->symbol() && m_reticle->symbol()->parent() == this)
+    {
+      // If we own the set symbol, delete it.
+      m_reticle->symbol()->deleteLater();
+    }
+
+    m_reticle->setSymbol(symbol);
+    emit symbolChanged();
+  }
+
+  /*!
+    \brief Returns the amount to scale the OverviewMap compared to the geoView.
+    Defaults to 25.0.
+   */
+  double OverviewMapController::scaleFactor() const
+  {
+    return m_scaleFactor;
+  }
+
+  /*!
+    \brief Sets the amount to scale the OverviewMap compared to the geoView.
+   */
+  void OverviewMapController::setScaleFactor(double scaleFactor)
+  {
+    if (m_scaleFactor == scaleFactor)
+      return;
+
+    m_scaleFactor = scaleFactor;
+    emit scaleFactorChanged();
+
+    // Force a redraw of the inset geometry.
+    if (auto sceneView = qobject_cast<SceneViewToolkit*>(m_geoView))
+    {
+      emit sceneView->viewpointChanged();
+    }
+    else if (auto mapView = qobject_cast<MapViewToolkit*>(m_geoView))
+    {
+      emit mapView->viewpointChanged();
+    }
+  }
+
+  /*!
+    \internal
+    Disables certain interactions on the insetView.
+   */
+  void OverviewMapController::handleInteractions()
+  {
+    // Disable all keyboard interactions
+    QObject::connect(m_insetView, &MapViewToolkit::keyPressed, this, [](QKeyEvent& e)
+                     {
+                       e.accept();
+                     });
+    QObject::connect(m_insetView, &MapViewToolkit::keyReleased, this, [](QKeyEvent& e)
+                     {
+                       e.accept();
+                     });
+    QObject::connect(m_insetView, &MapViewToolkit::mouseWheelChanged, this, [](QWheelEvent& e)
+                     {
+                       e.accept();
+                     });
+    // Disable all mouse interactions on devices.
+#if defined Q_OS_ANDROID || defined Q_OS_IOS
```

Comment:
Be sure to include `QtGlobal` in this file to be sure the platform defines are available.